src/main/java/com/musala/sdcs/device/DeviceImpl.java

```diff
@@ -99,15 +112,21 @@ public void setLabel(String label) {
         this.label = label;
     }
 
+    @Override
     public List<Channel> getChannels() {
         return channels;
     }
 
+    @Override
     public void setChannels(List<Channel> channels) {
         this.channels = channels;
     }
     
     public void addChannel(Channel channel) {
```

Comment:
If we set channels through the constructor, creating new ArrayList<Channel>() will be unneccessary instance creation. the case with if(this.channels != null) is for when we give the constructor null value for the channels. And initialization is done exatly once when needed. Small improvement but still it's not a problem at all.